lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -628,11 +746,12 @@ def _load_info(self, name) -> Snap:
         info = self._snap_client.get_snap_information(name)
 
         return Snap(
-            info["name"],
-            SnapState.Available,
-            info["channel"],
-            info["revision"],
-            info["confinement"],
+            name=info["name"],
+            state=SnapState.Available,
+            channel=info["channel"],
+            revision=info["revision"],
+            confinement=info["confinement"],
+            apps=None,
```

Comment:
I remember why I did it now.

`Snap.__repr__` would print `apps`, so I wanted it populated at `__init__`, without having to make an API call for every Snap that gets initialised in `SnapCache._load_installed_snaps` and `SnapCache._load_info`.

Do you have an idea on how to avoid that?

Reply:
Excellent, I'll ignore `repr` and let it store after reference to `.apps`!